### DISCUSSION TITLE
React Warning: Prop `width` did not match. Server: "250" Client: "500"

### DISCUSSION THREAD

**Discussion by colnick** [vercel/next.js#18379] [Help] [answered]
I'm getting the following error in React when using the Next.js framework. I've read numerous articles on Github and Stack Overflow which suggests modifying/adding the `.babelrc `file but I still have no luck. The strange thing is, when I run the code in a fresh install of Next.js it works but then shortly after stops? Is this issue caching related?

```
Warning: Prop `width` did not match. Server: "250" Client: "500"
in canvas (created by i)
in i (at upload.js:22)
in div (at upload.js:21)
in ImageCrop (at test.js:5)
in Test (at _app.js:65)
in App
in ErrorBoundary (created by ReactDevOverlay)
in ReactDevOverlay (created by Container)
in Container (created by AppContainer)
in AppContainer
in Root
```

Any help would be appreciated!

**Comment by jamesmosier** [comment]
This means that some value is changing between the client and server by some side effect (i.e. `useEffect` or something similar). Caching does not come into play with this.

Do you have a reproduction of this you could share?

**Reply by colnick** [reply]
@jamesmosier Thanks for commenting! I actually managed to resolve it by using the Next.js dynamic import:

`const ImageCrop = dynamic(() => import('../components/avatar-editor'), { ssr: false })
`

**Comment by colnick** [comment, marked as answer]
In case anyone else come across a similar issue, it turned out to be something to do with the Next.js server side rendering (SSR). I managed to resolve it by using the Next.js dynamic import:

`const ImageCrop = dynamic(() => import('../components/avatar-editor'), { ssr: false })`

You can read more about dynamic import in Next.js here: https://nextjs.org/docs/advanced-features/dynamic-import
